src/core/opamVersionCompare.ml

```diff
@@ -15,43 +15,27 @@
 let is_digit = function
   | '0'..'9' -> true
   | _ -> false
-;;
 
 (* [skip_while_from i f w m] yields the index of the leftmost character
  * in the string [s], starting from [i], and ending at [m], that does
  * not satisfy the predicate [f], or [length w] if no such index exists.  *)
 let skip_while_from i f w m =
   let rec loop i =
-    if i = m then i
+    if Int.equal i m then i
     else if f w.[i] then loop (i + 1) else i
   in loop i
-;;
-
-(* splits a version into (epoch,rest), without the separating ':'. The
- * epoch is delimited by the leftmost occurrence of ':' in x, and is ""
- * in case there is no ':' in x.  *)
-let extract_epoch x =
```

Comment:
Note for reviewers: This was removed because `OpamPackage.Version.of_string` disallows `:`

Reply:
That's good to know - getting rid of the remaining allocs should be a lot easier if we don't need to support epochs!